src/components/WorkBuddies/WorkBuddies.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {
+  Fragment,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useMemo,
+} from "react";
+import "./work-buddies.css";
+import classNames from "classnames";
+import { Buddy } from "./BuddyInterface";
+import { BuddiesContext } from "../BuddiesContext";
+import MyWorkBuddy from "./MyWorkBuddy";
+
+const WorkBuddies = () => {
+  const [buddies, setBuddies] = useState<Buddy[]>([]);
+  const [myBuddy, setMyBuddy] = useState<Buddy>();
+  const [isBuddyChosen, setIsBuddyChosen] = useState<boolean>(false);
+  //useRef
+  const nameRef = useRef<HTMLInputElement>(null);
+
+  //useContext
+  const { setData } = useContext(BuddiesContext);
+
+  //useCallback
+  const fetchBuddies = useCallback(async () => {
+    try {
+      const response = await fetch(
+        "https://pokeapi.co/api/v2/pokemon?limit=100"
+      );
+      const data = await response.json();
+
+      const randomIndexes: number[] = [];
+
+      while (randomIndexes.length < 9) {
+        const randomNumber = Math.floor(Math.random() * 100);
+        if (!randomIndexes.includes(randomNumber)) {
+          randomIndexes.push(randomNumber);
+        }
+      }
+
+      const todaysBuddies: Buddy[] = randomIndexes.map((index: number) => {
+        let urlSplitted = data.results[index].url.split("/");
+        let newUrl = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${
+          urlSplitted[urlSplitted.length - 2]
+        }.png`;
+
+        return { name: data.results[index].name, url: newUrl };
+      });
+
+      setBuddies(todaysBuddies);
+    } catch (error) {
+      console.log(error);
+    }
+  }, [setBuddies]);
+
+  //useMemo
+  const displayBuddies = useMemo(() => {
```

Comment:
```suggestion
  const buddiesToDisplayed = useMemo(() => {
```

Reply:
Is this a common naming convention when using useMemo?

Reply:
No. The idea is that this is something you want to display on the screen.
The best name would be just `buddies`. Since, that is already taken, you can use something like `buddiesToDisplay`